From: Hephy Builder <hephy@urmanac.com>
Date: Sat, 16 Nov 2025 00:00:00 +0000
Subject: [PATCH 1/3] Add ARM64 architecture support to Talos profile generator

This patch modifies the hack/gen-profiles.sh script to generate ARM64 profiles
instead of AMD64, making the build compatible with t4g instances and
Raspberry Pi CM3 modules.

Changes:
- Update arch field to arm64
- Change paths from /usr/install/amd64/ to /usr/install/arm64/
- Update output filenames to include arm64 suffix

---
 packages/core/installer/hack/gen-profiles.sh | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/packages/core/installer/hack/gen-profiles.sh b/packages/core/installer/hack/gen-profiles.sh
index 1234567..abcdefg 100755
--- a/packages/core/installer/hack/gen-profiles.sh
+++ b/packages/core/installer/hack/gen-profiles.sh
@@ -50,12 +50,12 @@ for profile in $PROFILES; do
   cat > images/talos/profiles/$profile.yaml <<EOT
 # this file generated by hack/gen-profiles.sh
 # do not edit it
-arch: amd64
+arch: arm64
 platform: ${platform}
 secureboot: false
 version: ${TALOS_VERSION}
 input:
   kernel:
-    path: /usr/install/amd64/vmlinuz
+    path: /usr/install/arm64/vmlinuz
   initramfs:
-    path: /usr/install/amd64/initramfs.xz
+    path: /usr/install/arm64/initramfs.xz
   baseInstaller:
     imageRef: "ghcr.io/siderolabs/installer:${TALOS_VERSION}"
   systemExtensions:
@@ -72,4 +72,4 @@ output:
   outFormat: ${out_format}
 EOT
 done
+
+# Update Makefile targets to use arm64 suffixes
+sed -i 's/amd64/arm64/g' ../Makefile